content/blog/the-pulumi-intern-experience/index.md

```diff
@@ -0,0 +1,59 @@
+
+### Asking Questions
+
+“There is no such thing as a dumb question” was a phrase I had been hearing a lot, but it never fully registered until recently. The “cloud” as we know it is quite vast. Setting it up takes hours, and fully understanding what is going on could take tenfold. Pulumi simplifies the process; however, there is still so much going on even when doing a single thing. This experience truly taught me that there really is no such thing as a dumb question and searching the internet to find answers is an expectation! Nobody knows all the answers, and that is completely okay.
+
+### Remote Lifestyle
+
+I’m definitely not a fan of the [remote environment](https://www.pulumi.com/blog/coronavirus-plan/), however, I’m the kind of person that is more productive at night, so it allowed me to shift my work to times when I knew I would get things done. That said, face to face interaction was something I missed a lot, as it’s understandably harder to get to know people over a computer. However, since Pulumi already has many remote engineers, it was a very smooth process to get set up and start work right away.
+
+## The Project Experience
+
+### Introduction to the Pulumi Model
+
+I worked on the Platform team for the last 3 months, and they focused their work on the Pulumi SDK. The first few weeks were a huge learning curve, as the interns were tasked with converting Pulumi deployment examples from Typescript to Python, C#, and Go. This intro project taught me a lot about cloud resources and how they are deployed and helpful debugging tools through this process. I also learned how to write [tests](https://www.pulumi.com/docs/guides/testing/) to validate output for a deployment which proved helpful as I learned how the entire process was working.
+
+### kube2pulumi Tooling
+
+I built a CLI tool, and an open-source library to convert Kubernetes YAML manifests to Pulumi in a user-defined language ([check out kube2pulumi!](https://www.pulumi.com/blog/introducing-kube2pulumi/). This was done so that Kubernetes users could easily migrate to Pulumi, and use familiar programming languages to tame YAML's complexity. I truly learned a lot from this experience as it was my first time designing and implementing a project end to end in a set time frame. Utilizing existing open-source libraries, code generation techniques, and compilers theory were just some of the many things I got to delve into.
```

Comment:
```suggestion
I built a CLI tool and an open-source library to convert Kubernetes YAML manifests to Pulumi in a user-defined language ([check out kube2pulumi!]({{< relref "/blog/introducing-kube2pulumi" >}}). This was done so that Kubernetes users could easily migrate to Pulumi and use familiar programming languages to tame YAML's complexity. I truly learned a lot from this experience as it was my first time designing and implementing a project end to end in a set time frame. Utilizing existing open-source libraries, code generation techniques, and compilers theory were just some of the many things I got to delve into.
```